### PR TITLE
TFS-06 block synchronization testing

### PR DESCRIPTION
**Pull request by alanchchen**

### CODE REVIEW

File: tests/block_sync_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+			container.NAT("any"),
+			container.NoDiscover(),
+			container.Etherbase("1a9afb711302c5f83b5902843d1c007a1a137632"),
+			container.Mine(),
+		)
+
+		Expect(blockchain.Start()).To(BeNil())
+	})
+
+	AfterSuite(func() {
```
Comment (yutelin) [inline tests/block_sync_test.go]: Should we use `AfterSuite` or `AfterEach` here?

File: tests/block_sync_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+				for _, v := range blockchain.Validators() {
+					nodeClient.AddPeer(context.Background(), v.NodeAddress())
+				}
+
+				nodeClient.Close()
+			}
+
+			By("Wait for p2p connection")
+			<-time.After(10 * time.Second)
```
Comment (yutelin) [inline tests/block_sync_test.go]: A way to improve the speed of testing, we can poll blockchain.Validators() like every 100 millisecond and set a timeout for the whole `It`

Following will at most run for 10 seconds.
```
It("abc", func(done Done){
 close(done);
}, 10);
```

File: tests/block_sync_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+				Expect(err).To(BeNil())
+				Expect(len(p2pInfos)).To(Equal(len(blockchain.Validators())))
+
+				nodeClient.Close()
+			}
+		})
+
+		It("TFS-06-02 node synchronization", func() {
+			By("Wait for block generation")
+			<-time.After(10 * time.Second)
```
Comment (yutelin) [inline tests/block_sync_test.go]: same comment as above.

File: tests/block_sync_test.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+				Expect(err).To(BeNil())
+				Expect(len(p2pInfos)).To(Equal(len(blockchain.Validators())))
+
+				nodeClient.Close()
+			}
+		})
+
+		It("TFS-06-02 node synchronization", func() {
+			By("Wait for block generation")
```
Comment (bailantaotao) [inline tests/block_sync_test.go]: Use `blockchain.ConsensusMonitor`?

File: tests/block_sync_test.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+			for _, v := range blockchain.Validators() {
+				client := v.NewIstanbulClient()
+				Expect(client).NotTo(BeNil())
+				err := client.StopMining(context.Background())
+				Expect(err).To(BeNil())
+				client.Close()
+			}
+
+			By("Wait for block synchronization between validators")
+			waitFor(blockchain.Validators(), func(geth container.Ethereum, wg *sync.WaitGroup) {
```
Comment (yutelin) [inline tests/block_sync_test.go]: Will this always pass because `WaitForBlocks` was run before this?

**Pull request merged**